src/main/scala/castalia/matcher/MatchResultGatherer.scala

```diff
@@ -0,0 +1,62 @@
+package castalia.matcher
+
+import akka.actor._
+import akka.http.scaladsl.model.StatusCodes._
+import akka.pattern.{ask, pipe}
+import akka.util.Timeout
+import castalia.matcher.MatchResultGatherer.{MatchFound, MatchNotFound}
+import castalia.model.Model.StubResponse
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+
+/**
+  * Created by Jordi on 4-2-2016.
+  */
+object MatchResultGatherer {
+
+  case object MatchNotFound
+
+  case class MatchFound(stubHandler: ActorRef, requestMatch: RequestMatch)
+
+  def props(nOfMatchers: Int, origin: ActorRef): Props = Props(new MatchResultGatherer(nOfMatchers, origin))
+}
+
+class MatchResultGatherer(nrOfMatchers: Int, origin: ActorRef) extends Actor with ActorLogging {
+  implicit val timeout = Timeout(2.seconds)
```

Comment:
The user will get a Failure returned, which kind of makes sense since something went wrong inside the stubserver.
I think this timeout should be equal to a global timeout that applies to the entire stubserver. Changing it to be equal to Main.timeout, or just importing Main.timeout, leads to failing unit tests.